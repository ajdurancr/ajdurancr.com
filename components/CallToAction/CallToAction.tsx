import React from 'react'
import get from 'lodash.get'

type Cta = {
  text: string,
  link: string,
}

const CallToAction: React.FC<{ cta: Cta }> = ({ cta }) => (
  <a href={get(cta, 'link')}>{get(cta, 'text')}</a>
)

export { CallToAction }
