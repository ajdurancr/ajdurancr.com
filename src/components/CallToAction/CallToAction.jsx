import React from 'react'
import propTypes from 'prop-types'
import get from 'lodash.get'

const CallToAction = ({ cta }) => (
  <a href={get(cta, 'link')}>{get(cta, 'text')}</a>
)

CallToAction.propTypes = {
  cta: propTypes.shape({
    link: propTypes.string.isRequired,
    text: propTypes.string.isRequired,
  }).isRequired,
}

export default CallToAction
