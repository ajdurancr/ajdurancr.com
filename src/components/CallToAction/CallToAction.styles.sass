@import '../../styles/_variables'
  
.header
  display: flex
  flex-direction: column
  left: 0 
  position: fixed
  top: 0
  width: 100%
  margin: 0 3.2rem

  @media only screen and (min-width: $size-tablet)
    margin: 0 6.4rem

.topMenu
  display: flex
  justify-content: space-between
  min-height: 5rem
  align-items: center
  padding: 1.6rem 0

  .logo
    display: flex
    max-width: 15rem

    @media only screen and (min-width: $size-tablet)
      max-width: 20rem

