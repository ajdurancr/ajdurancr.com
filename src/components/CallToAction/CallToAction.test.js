import React from 'react'
import { shallow } from 'enzyme'

import CallToAction from './CallToAction'

describe('CallToAction', () => {
  it('renders correctly', () => {
    const cta = { link: '/test/link', text: 'Test link' }
    const wrapper = shallow(<CallToAction cta={cta} />)

    expect(wrapper).toMatchSnapshot()
  })
})
