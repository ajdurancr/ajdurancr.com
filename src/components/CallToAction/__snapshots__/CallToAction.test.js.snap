// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallToAction renders correctly 1`] = `
<a
  href="/test/link"
>
  Test link
</a>
`;
